manual/thread_safetiness/index.md

```diff
@@ -0,0 +1,23 @@
+---
+title: Thread safetiness
+layout: single
+sidebar:
+  nav: "manual"
+toc: true
+toc_sticky: true
+---
+
+ROOT classes support one of the following thread-safety levels:
+
+- Thread unsafe: objects must not be used concurrently by different threads, even if objects are not shared among threads.
+
+- Conditionally thread safe: objects can be used safely by multiple threads concurrently as long as mutiple threads do not share the same object (i.e. every thread uses a private copy of its objects).  In addition, objects can be shared among threads as long as all threads use only `const` methods of the shared objects.
+
+- Thread safe: objects can safely be shared and used concurrently by multiple threads.
+
+Unless otherwise stated, classes starting with `R` (e.g. `RDataFrame`) are conditionally thread safe.
+Classes starting with `T` (e.g. `TDirectory`) are thread unsafe.
```

Comment:
maybe:

```suggestion
Types whose name starts with `T` (e.g. `TDirectory`) as well as all globals (such as `gROOT`) are thread unsafe.
```

Reply:
TDirectory is conditionally thread safe.  gROOT is thread safe. :)

Reply:
We should add this information to the class documentation of `TDirectory` and `gROOT`.